conans/client/tools/scm.py

```diff
@@ -61,15 +61,58 @@ def run(self, command):
                     else:
                         return self._runner(command)
 
+    def handle_scp_pattern(self, user, domain, url):
+        if self._password:
+            self._output.warn("SCM password cannot be set for scp url, ignoring parameter")
+        if self._username:
+            self._output.warn("SCM username got from URL, ignoring 'username' parameter")
+        return "{user}@{domain}:{url}".format(user=user, domain=domain, url=url)
+
+    def handle_url_pattern(self, scheme, url, user=None, password=None):
```

Comment:
If not local to the function `get_url_with_credentials`, at least make them _private_

```suggestion
    def _handle_url_pattern(self, scheme, url, user=None, password=None):
```